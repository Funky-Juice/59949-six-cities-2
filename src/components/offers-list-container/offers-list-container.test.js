import renderer from 'react-test-renderer';
import OffersListContainer from './offers-list-container';
import offerTestObj from '../../mocks/test-offer';
import {Provider} from 'react-redux';
import {createStore} from 'redux';

const mockOffers = [offerTestObj];

const store = createStore(() => ({
  activeCity: mockOffers[0].city,
  activeOffers: mockOffers
}));

it(`OffersListContainer correctly renders`, () => {
  const tree = renderer
    .create(<Provider store={store}>
      <OffersListContainer
        activeCity={mockOffers[0].city}
        activeOffers={mockOffers}
      />
    </Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
