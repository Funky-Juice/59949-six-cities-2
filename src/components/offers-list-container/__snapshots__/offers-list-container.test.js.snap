// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OffersListContainer correctly renders 1`] = `
<section
  className="cities__places places"
>
  <b
    className="places__found"
  >
    1
     
    place
     to stay in 
    Amsterdam
  </b>
  <div
    className="cities__places-list places__list tabs__content"
  >
    <article
      className="cities__place-card place-card"
      onMouseEnter={[Function]}
    >
      <div
        className="cities__image-wrapper place-card__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="200"
            src="img/room.jpg"
            width="260"
          />
        </a>
      </div>
      <div
        className="place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              80
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button place-card__bookmark-button--active button"
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              In bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            >
              3
            </span>
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
          onClick={[Function]}
        >
          <a
            href="#"
          >
            Wood and stone place
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          Room
        </p>
      </div>
    </article>
  </div>
</section>
`;

exports[`OffersListContainer correctly renders without any offers 1`] = `
<section
  className="cities__no-places"
>
  <div
    className="cities__status-wrapper tabs__content"
  >
    <b
      className="cities__status"
    >
      No places to stay available
    </b>
    <p
      className="cities__status-description"
    >
      We could not find any property availbale
    </p>
  </div>
</section>
`;
